src/Microsoft.DocAsCode.MarkdownLite/JsonTokenTreeRenderer.cs

```diff
@@ -219,58 +226,70 @@ public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownTextToken
 
         public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownIgnoreToken token, IMarkdownContext context)
         {
-            return this.Insert("Ignore", this.GetSize(), Type.Leaf);
+            return Insert(token, $"Ignore>{escape(token.SourceInfo.Markdown)}");
         }
 
         public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownRawToken token, IMarkdownContext context)
         {
-            return this.Insert($"Raw(From{token.Rule.Name})", GetSize(), Type.Leaf);
+            return Insert(token, $"Raw(From{token.Rule.Name})>{escape(token.SourceInfo.Markdown)}");
         }
 
         #endregion
 
         #region Private
 
-        private enum Type
+        protected StringBuffer Insert(IMarkdownToken token, StringBuffer name, StringBuffer tokenContent = null)
         {
-            Leaf,
-            NonLeaf
+            int startLineNumber = token.SourceInfo.LineNumber;
+            var content = token.SourceInfo.Markdown;
+            int indexOfLastChar = content.Length - 1;
+            while (indexOfLastChar >= 0 && content[indexOfLastChar] == '\n')
+                indexOfLastChar--;
+            int endLineNumber = startLineNumber + content.Substring(0, indexOfLastChar + 1).Count(ch => ch == '\n');
+            StringBuffer result = $"{{\"name\":\"{startLineNumber}>{endLineNumber}>{name}\"";
```

Comment:
for that the node in d3.js can only has the 'name','children' and 'size' properties

---
In reply to: [76182807](https://github.com/dotnet/docfx/pull/631#discussion_r76182807) [](ancestors = 76182807)